From e8fe173ab3a0740e86c9a446aba223c1f8271e77 Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Tue, 10 Sep 2024 13:36:25 +0300
Subject: [PATCH] [FSW-6362] pkg: added package changes for mgmt io

Added config and make file changes to build for
management IO.
Now we have two configs one for gaudi products and
one for GPU.

Change-Id: If7e9c02c0a9109c5174d2c27cc58f75147afaa31
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/package/mgmt/Config.in b/package/mgmt/Config.in
index 5591bfe..c32d799 100644
--- a/package/mgmt/Config.in
+++ b/package/mgmt/Config.in
@@ -2,3 +2,15 @@
 	bool "Zephyr ARC MGMT"
 	help
 	  Will be used to handle MGMT on secondary CPU
+
+config BR2_PACKAGE_MGMT_IO
+	bool "Zephyr ARC MGMT IO"
+	select BR2_PACKAGE_MGMT
+	help
+	  Will be used to handle management arc for IO die
+
+config BR2_PACKAGE_MGMT_GAUDI
+	bool "Zephyr ARC MGMT for gaudi products"
+	select BR2_PACKAGE_MGMT
+	help
+	  Will be used to handle MGMT for gaudi
diff --git a/package/mgmt/mgmt.mk b/package/mgmt/mgmt.mk
index 93173c5..2dea3be 100644
--- a/package/mgmt/mgmt.mk
+++ b/package/mgmt/mgmt.mk
@@ -64,7 +64,13 @@
 		$(MGMT_CONF_CFLAGS)"
 
 MGMT_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMB_PROJ_DIR)"
+
+ifdef BR2_PACKAGE_MGMT_IO
+MGMT_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_$(MGMT_APP_NAME)_io_$(HL_TARGET)
+else
 MGMT_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_$(MGMT_APP_NAME)_$(HL_TARGET)
+endif
+
 MGMT_CONF_OPTS += -DDEVICE=$(MGMT_DEVICE_NAME)
 MGMT_CONF_OPTS += -DAPP_NAME=$(MGMT_APP_PROD_NAME)
 MGMT_CONF_OPTS += -DAPP_VERSION=$(MGMT_APP_VERSION)
@@ -135,10 +141,18 @@
 	@$(TARGET_MAKE_ENV) $(MAKE) -C $(MGMT_BUILD_DIR)/build all
 endef
 
+ifdef BR2_PACKAGE_MGMT_IO
 define MGMT_INSTALL_TARGET_CMDS
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(ELFS_DIR)/zephyr_$(MGMT_APP_NAME).elf
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(STATS_DIR)/zephyr_$(MGMT_APP_NAME).stat
+@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(ELFS_DIR)/zephyr_$(MGMT_APP_NAME)_io.elf
+@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(STATS_DIR)/zephyr_$(MGMT_APP_NAME)_io.stat
 endef
+else
+define MGMT_INSTALL_TARGET_CMDS
+@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(ELFS_DIR)/zephyr_$(MGMT_APP_NAME).elf
+@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(STATS_DIR)/zephyr_$(MGMT_APP_NAME).stat
+endef
+endif
+
 
 mgmt-configure: $(MGMT_DEPENDENCIES)
 mgmt-menuconfig mgmt-guiconfig: | mgmt-configure
